worker/dbaccessor/tracker_test.go

```diff
@@ -0,0 +1,309 @@
+	case <-time.After(testing.ShortWait):
+		c.Fatal("timed out waiting for DB callback")
+	}
+
+	tables := checkTableNames(c, w)
+	c.Assert(tables, SliceContains, "lease")
+
+	workertest.CleanKill(c, w)
+
+	c.Assert(w.Err(), jc.ErrorIsNil)
+}
+
+func (s *trackedDBWorkerSuite) TestWorkerAttemptsToVerifyDBButSucceedsWithDifferentDB(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.expectAnyLogs()
+	s.expectClock()
+	done := s.expectTimer(1)
+
+	s.logger.EXPECT().Errorf(gomock.Any(), gomock.Any()).AnyTimes()
+
+	dbChange := make(chan struct{})
+	exp := s.dbApp.EXPECT()
+	gomock.InOrder(
+		exp.Open(gomock.Any(), "controller").Return(s.DB(), nil),
+		exp.Open(gomock.Any(), "controller").Return(s.DB(), nil),
+		exp.Open(gomock.Any(), "controller").DoAndReturn(func(_ context.Context, _ string) (*sql.DB, error) {
+			defer close(dbChange)
+			return s.NewCleanDB(c), nil
+		}),
+	)
+
+	var count uint64
+	verifyFn := func(context.Context, *sql.DB) error {
+		val := atomic.AddUint64(&count, 1)
+
+		if val == DefaultVerifyAttempts {
+			return nil
+		}
+		return errors.New("boom")
+	}
+
+	w, err := NewTrackedDBWorker(s.dbApp, "controller", WithClock(s.clock), WithLogger(s.logger), WithVerifyDBFunc(verifyFn))
+	c.Assert(err, jc.ErrorIsNil)
+
+	defer workertest.DirtyKill(c, w)
+
+	select {
+	case <-done:
+	case <-time.After(testing.ShortWait):
+		c.Fatal("timed out waiting for DB callback")
+	}
+
+	// The db should have changed to the new db.
+	select {
+	case <-dbChange:
+	case <-time.After(testing.ShortWait):
+		c.Fatal("timed out waiting for DB callback")
+	}
+
+	tables := checkTableNames(c, w)
+	c.Assert(tables, gc.Not(SliceContains), "lease")
+
+	workertest.CleanKill(c, w)
+
+	c.Assert(w.Err(), jc.ErrorIsNil)
+}
+
+func (s *trackedDBWorkerSuite) TestWorkerAttemptsToVerifyDBButFails(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.expectAnyLogs()
+	s.expectClock()
+	done := s.expectTimer(1)
+
+	s.logger.EXPECT().Errorf(gomock.Any(), gomock.Any()).AnyTimes()
+
+	s.dbApp.EXPECT().Open(gomock.Any(), "controller").Return(s.DB(), nil).Times(DefaultVerifyAttempts)
+
+	verifyFn := func(context.Context, *sql.DB) error {
+		return errors.New("boom")
+	}
+
+	w, err := NewTrackedDBWorker(s.dbApp, "controller", WithClock(s.clock), WithLogger(s.logger), WithVerifyDBFunc(verifyFn))
+	c.Assert(err, jc.ErrorIsNil)
+
+	defer workertest.DirtyKill(c, w)
+
+	select {
+	case <-done:
+	case <-time.After(testing.ShortWait):
+		c.Fatal("timed out waiting for DB callback")
+	}
+
+	c.Assert(w.Wait(), gc.ErrorMatches, "boom")
+	c.Assert(w.Err(), gc.ErrorMatches, "boom")
+
+	// Ensure that the DB is dead.
+	err = w.DB(func(db *sql.DB) error {
+		c.Fatal("failed if called")
+		return nil
+	})
+	c.Assert(err, gc.ErrorMatches, "boom")
+	err = w.Txn(context.TODO(), func(ctx context.Context, tx *sql.Tx) error {
+		c.Fatal("failed if called")
+		return nil
+	})
+	c.Assert(err, gc.ErrorMatches, "boom")
+}
+
+func checkTableNames(c *gc.C, w coredatabase.TrackedDB) []string {
```

Comment:
is this 'checkTableNames' or just 'readTableNames' ?
There isn't a lot of checking of their content here.
Which also seems to say that the 'should be any leases' comment should be in the caller, not in this function.